modules/fuzzy/include/opencv2/fuzzy/fuzzy_F1_math.hpp

```diff
@@ -0,0 +1,152 @@
+    @param matrix Input array.
+    @param kernel Kernel used for processing. Function **createKernel** can be used.
+    @param c00 Elements represent average color.
+    @param c10 Elements represent average vertical gradient.
+    @param c01 Elements represent average horizontal gradient.
+    @param components Output 32-bit array for the components.
+    @param mask Mask can be used for unwanted area marking.
+
+    The function computes components and its elements using predefined kernel and mask.
+
+    @note
+        F-transform technique of first degreee is described in paper @cite Vlas:FT.
+    */
+    CV_EXPORTS_AS(FT12D_polynomial1) void FT12D_polynomial(InputArray matrix, InputArray kernel, OutputArray c00, OutputArray c10, OutputArray c01, OutputArray components, InputArray mask);
+
+    /** @brief Computes elements of F1-transform components.
+    @param matrix Input array.
+    @param kernel Kernel used for processing. Function **createKernel** can be used.
+    @param c00 Elements represent average color.
+    @param c10 Elements represent average vertical gradient.
+    @param c01 Elements represent average horizontal gradient.
+    @param components Output 32-bit array for the components.
+
+    The function computes components and its elements using predefined kernel.
+
+    @note
+        F-transform technique of first degreee is described in paper @cite Vlas:FT.
+    */
+    CV_EXPORTS_W void FT12D_polynomial(InputArray matrix, InputArray kernel, OutputArray c00, OutputArray c10, OutputArray c01, OutputArray components);
+
+    /** @brief Creates vertical matrix for F1-transform computation.
+    @param radius Radius of the basic function.
+    @param matrix The vertical matrix.
+    @param chn Number of channels.
+
+    The function creates helper vertical matrix for F1-transfrom processing. It is used for gradient computation.
+    */
+    CV_EXPORTS_W void FT12D_createPolynomMatrixVertical(int radius, OutputArray matrix, const int chn);
+
+    /** @brief Creates horizontal matrix for F1-transform computation.
+    @param radius Radius of the basic function.
+    @param matrix The horizontal matrix.
+    @param chn Number of channels.
+
+    The function creates helper horizontal matrix for F1-transfrom processing. It is used for gradient computation.
+    */
+    CV_EXPORTS_W void FT12D_createPolynomMatrixHorizontal(int radius, OutputArray matrix, const int chn);
+
+    /** @brief Computes F1-transfrom and inverse F1-transfrom at once.
+    @param matrix Input matrix.
+    @param kernel Kernel used for processing. Function **createKernel** can be used.
+    @param output Output 32-bit array.
+
+    This function computes F1-transfrom and inverse F1-transfotm in one step. It is fully sufficient and optimized for **Mat**.
+    */
+    CV_EXPORTS_W void FT12D_process(InputArray matrix, InputArray kernel, OutputArray output);
+
+    /** @brief Computes F1-transfrom and inverse F1-transfrom at once.
```

Comment:
How it's possible to compute direct and inverse transform at once, should it have two output parameters for both transforms? Or type of transform depends on passed kernel?